### ISSUE TITLE
The type of `timeZone` could be `undefined` but it's marked as a `string` only instead

### ISSUE THREAD

**Issue by luciomartinez** [microsoft/TypeScript#56057]
### ⚙ Compilation target

es5

### ⚙ Library

Intl

### Missing / Incorrect Definition

The type of `Intl.DateTimeFormat.prototype.resolvedOptions().timeZone` is incorrectly set to only `string` where it could also be `undefined`.

### Sample Code

```TypeScript
// TypeScript should raise an error in the next line since the return value can be `undefined`
const getTimeZone = (): string => {
  return new Intl.DateTimeFormat().resolvedOptions().timeZone;
};
```

### Documentation Link

See possible values for the `timeZone` property in [MDN `resolvedOptions()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/resolvedOptions#description), where `undefined` is included.

**Comment by jakebailey** [comment]
https://github.com/microsoft/TypeScript/issues/55869 ?

**Comment by fatcerberus** [comment]
See https://github.com/microsoft/TypeScript/issues/55869#issuecomment-1736456729.
`string` is the correct type according to the specification; Chrome is apparently operating out of spec.

**Comment by luciomartinez** [comment]
> See #55869, the spec says this should be a string, no?

Really good catch! I've a few questions left so I've [posted them there](https://github.com/microsoft/TypeScript/issues/55869#issuecomment-1755710962).:)

> Chrome is apparently operating out of spec.

You're right @fatcerberus ! Chrome is still returning `undefined` in some cases, so it's either a bug on their end or misleading [definitions in ECMAScript](https://tc39.es/ecma402/#sec-intl.datetimeformat.prototype.resolvedoptions).:)

Closing this issue as duplicated now. Thanks
